### ISSUE TITLE
Call to undefined method Psr\Log\NullLogger::channel() when Running CLI

### ISSUE THREAD

**Issue by alexclst** [deliciousbrains/spinupwp-cli#54]
## Current Outcome
I cannot run any of the Spinupwp CLI since I get the following errors:

```
PHP Deprecated:  Return type of HumbugBox3130\KevinGH\RequirementChecker\RequirementCollection::getIterator() should either be compatible with IteratorAggregate::getIterator(): Traversable, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in phar:///Users/alex/.composer/vendor/spinupwp/spinupwp-cli/builds/spinupwp/.box/src/RequirementCollection.php on line 12

Deprecated: Return type of HumbugBox3130\KevinGH\RequirementChecker\RequirementCollection::getIterator() should either be compatible with IteratorAggregate::getIterator(): Traversable, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in phar:///Users/alex/.composer/vendor/spinupwp/spinupwp-cli/builds/spinupwp/.box/src/RequirementCollection.php on line 12
PHP Deprecated:  Return type of HumbugBox3130\KevinGH\RequirementChecker\RequirementCollection::count() should either be compatible with Countable::count(): int, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in phar:///Users/alex/.composer/vendor/spinupwp/spinupwp-cli/builds/spinupwp/.box/src/RequirementCollection.php on line 16

Deprecated: Return type of HumbugBox3130\KevinGH\RequirementChecker\RequirementCollection::count() should either be compatible with Countable::count(): int, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in phar:///Users/alex/.composer/vendor/spinupwp/spinupwp-cli/builds/spinupwp/.box/src/RequirementCollection.php on line 16
PHP Deprecated:  Return type of Illuminate\Container\Container::offsetExists($key) should either be compatible with ArrayAccess::offsetExists(mixed $offset): bool, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in phar:///Users/alex/.composer/vendor/spinupwp/spinupwp-cli/builds/spinupwp/vendor/illuminate/container/Container.php on line 1409

Deprecated: Return type of Illuminate\Container\Container::offsetExists($key) should either be compatible with ArrayAccess::offsetExists(mixed $offset): bool, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in phar:///Users/alex/.composer/vendor/spinupwp/spinupwp-cli/builds/spinupwp/vendor/illuminate/container/Container.php on line 1409
PHP Deprecated:  Return type of Illuminate\Container\Container::offsetGet($key) should either be compatible with ArrayAccess::offsetGet(mixed $offset): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in phar:///Users/alex/.composer/vendor/spinupwp/spinupwp-cli/builds/spinupwp/vendor/illuminate/container/Container.php on line 1421

Deprecated: Return type of Illuminate\Container\Container::offsetGet($key) should either be compatible with ArrayAccess::offsetGet(mixed $offset): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in phar:///Users/alex/.composer/vendor/spinupwp/spinupwp-cli/builds/spinupwp/vendor/illuminate/container/Container.php on line 1421
PHP Deprecated:  Return type of Illuminate\Container\Container::offsetSet($key, $value) should either be compatible with ArrayAccess::offsetSet(mixed $offset, mixed $value): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in phar:///Users/alex/.composer/vendor/spinupwp/spinupwp-cli/builds/spinupwp/vendor/illuminate/container/Container.php on line 1434

Deprecated: Return type of Illuminate\Container\Container::offsetSet($key, $value) should either be compatible with ArrayAccess::offsetSet(mixed $offset, mixed $value): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in phar:///Users/alex/.composer/vendor/spinupwp/spinupwp-cli/builds/spinupwp/vendor/illuminate/container/Container.php on line 1434
PHP Deprecated:  Return type of Illuminate\Container\Container::offsetUnset($key) should either be compatible with ArrayAccess::offsetUnset(mixed $offset): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in phar:///Users/alex/.composer/vendor/spinupwp/spinupwp-cli/builds/spinupwp/vendor/illuminate/container/Container.php on line 1448

Deprecated: Return type of Illuminate\Container\Container::offsetUnset($key) should either be compatible with ArrayAccess::offsetUnset(mixed $offset): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in phar:///Users/alex/.composer/vendor/spinupwp/spinupwp-cli/builds/spinupwp/vendor/illuminate/container/Container.php on line 1448
PHP Deprecated:  Return type of Illuminate\Config\Repository::offsetExists($key) should either be compatible with ArrayAccess::offsetExists(mixed $offset): bool, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in phar:///Users/alex/.composer/vendor/spinupwp/spinupwp-cli/builds/spinupwp/vendor/illuminate/config/Repository.php on line 142

Deprecated: Return type of Illuminate\Config\Repository::offsetExists($key) should either be compatible with ArrayAccess::offsetExists(mixed $offset): bool, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in phar:///Users/alex/.composer/vendor/spinupwp/spinupwp-cli/builds/spinupwp/vendor/illuminate/config/Repository.php on line 142
PHP Deprecated:  Return type of Illuminate\Config\Repository::offsetGet($key) should either be compatible with ArrayAccess::offsetGet(mixed $offset): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in phar:///Users/alex/.composer/vendor/spinupwp/spinupwp-cli/builds/spinupwp/vendor/illuminate/config/Repository.php on line 154

Deprecated: Return type of Illuminate\Config\Repository::offsetGet($key) should either be compatible with ArrayAccess::offsetGet(mixed $offset): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in phar:///Users/alex/.composer/vendor/spinupwp/spinupwp-cli/builds/spinupwp/vendor/illuminate/config/Repository.php on line 154
PHP Deprecated:  Return type of Illuminate\Config\Repository::offsetSet($key, $value) should either be compatible with ArrayAccess::offsetSet(mixed $offset, mixed $value): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in phar:///Users/alex/.composer/vendor/spinupwp/spinupwp-cli/builds/spinupwp/vendor/illuminate/config/Repository.php on line 167

Deprecated: Return type of Illuminate\Config\Repository::offsetSet($key, $value) should either be compatible with ArrayAccess::offsetSet(mixed $offset, mixed $value): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in phar:///Users/alex/.composer/vendor/spinupwp/spinupwp-cli/builds/spinupwp/vendor/illuminate/config/Repository.php on line 167
PHP Deprecated:  Return type of Illuminate\Config\Repository::offsetUnset($key) should either be compatible with ArrayAccess::offsetUnset(mixed $offset): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in phar:///Users/alex/.composer/vendor/spinupwp/spinupwp-cli/builds/spinupwp/vendor/illuminate/config/Repository.php on line 179

Deprecated: Return type of Illuminate\Config\Repository::offsetUnset($key) should either be compatible with ArrayAccess::offsetUnset(mixed $offset): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in phar:///Users/alex/.composer/vendor/spinupwp/spinupwp-cli/builds/spinupwp/vendor/illuminate/config/Repository.php on line 179
PHP Deprecated:  Return type of Symfony\Component\Finder\Finder::getIterator() should either be compatible with IteratorAggregate::getIterator(): Traversable, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in phar:///Users/alex/.composer/vendor/spinupwp/spinupwp-cli/builds/spinupwp/vendor/symfony/finder/Finder.php on line 611

Deprecated: Return type of Symfony\Component\Finder\Finder::getIterator() should either be compatible with IteratorAggregate::getIterator(): Traversable, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in phar:///Users/alex/.composer/vendor/spinupwp/spinupwp-cli/builds/spinupwp/vendor/symfony/finder/Finder.php on line 611
PHP Deprecated:  Return type of Symfony\Component\Finder\Finder::count() should either be compatible with Countable::count(): int, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in phar:///Users/alex/.composer/vendor/spinupwp/spinupwp-cli/builds/spinupwp/vendor/symfony/finder/Finder.php on line 694

Deprecated: Return type of Symfony\Component\Finder\Finder::count() should either be compatible with Countable::count(): int, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in phar:///Users/alex/.composer/vendor/spinupwp/spinupwp-cli/builds/spinupwp/vendor/symfony/finder/Finder.php on line 694
PHP Deprecated:  Return type of Symfony\Component\Finder\Iterator\FileTypeFilterIterator::accept() should either be compatible with FilterIterator::accept(): bool, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in phar:///Users/alex/.composer/vendor/spinupwp/spinupwp-cli/builds/spinupwp/vendor/symfony/finder/Iterator/FileTypeFilterIterator.php on line 45

Deprecated: Return type of Symfony\Component\Finder\Iterator\FileTypeFilterIterator::accept() should either be compatible with FilterIterator::accept(): bool, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in phar:///Users/alex/.composer/vendor/spinupwp/spinupwp-cli/builds/spinupwp/vendor/symfony/finder/Iterator/FileTypeFilterIterator.php on line 45
PHP Deprecated:  Return type of Symfony\Component\Finder\Iterator\RecursiveDirectoryIterator::hasChildren($allowLinks = false) should either be compatible with RecursiveDirectoryIterator::hasChildren(bool $allowLinks = false): bool, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in phar:///Users/alex/.composer/vendor/spinupwp/spinupwp-cli/builds/spinupwp/vendor/symfony/finder/Iterator/RecursiveDirectoryIterator.php on line 87

Deprecated: Return type of Symfony\Component\Finder\Iterator\RecursiveDirectoryIterator::hasChildren($allowLinks = false) should either be compatible with RecursiveDirectoryIterator::hasChildren(bool $allowLinks = false): bool, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in phar:///Users/alex/.composer/vendor/spinupwp/spinupwp-cli/builds/spinupwp/vendor/symfony/finder/Iterator/RecursiveDirectoryIterator.php on line 87
PHP Deprecated:  Return type of Symfony\Component\Finder\Iterator\RecursiveDirectoryIterator::getChildren() should either be compatible with RecursiveDirectoryIterator::getChildren(): RecursiveDirectoryIterator, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in phar:///Users/alex/.composer/vendor/spinupwp/spinupwp-cli/builds/spinupwp/vendor/symfony/finder/Iterator/RecursiveDirectoryIterator.php on line 111

Deprecated: Return type of Symfony\Component\Finder\Iterator\RecursiveDirectoryIterator::getChildren() should either be compatible with RecursiveDirectoryIterator::getChildren(): RecursiveDirectoryIterator, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in phar:///Users/alex/.composer/vendor/spinupwp/spinupwp-cli/builds/spinupwp/vendor/symfony/finder/Iterator/RecursiveDirectoryIterator.php on line 111
PHP Deprecated:  Return type of Symfony\Component\Finder\Iterator\RecursiveDirectoryIterator::rewind() should either be compatible with FilesystemIterator::rewind(): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in phar:///Users/alex/.composer/vendor/spinupwp/spinupwp-cli/builds/spinupwp/vendor/symfony/finder/Iterator/RecursiveDirectoryIterator.php on line 137

Deprecated: Return type of Symfony\Component\Finder\Iterator\RecursiveDirectoryIterator::rewind() should either be compatible with FilesystemIterator::rewind(): void, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in phar:///Users/alex/.composer/vendor/spinupwp/spinupwp-cli/builds/spinupwp/vendor/symfony/finder/Iterator/RecursiveDirectoryIterator.php on line 137
PHP Deprecated:  Return type of Symfony\Component\Finder\Iterator\RecursiveDirectoryIterator::current() should either be compatible with FilesystemIterator::current(): SplFileInfo|FilesystemIterator|string, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in phar:///Users/alex/.composer/vendor/spinupwp/spinupwp-cli/builds/spinupwp/vendor/symfony/finder/Iterator/RecursiveDirectoryIterator.php on line 62

Deprecated: Return type of Symfony\Component\Finder\Iterator\RecursiveDirectoryIterator::current() should either be compatible with FilesystemIterator::current(): SplFileInfo|FilesystemIterator|string, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in phar:///Users/alex/.composer/vendor/spinupwp/spinupwp-cli/builds/spinupwp/vendor/symfony/finder/Iterator/RecursiveDirectoryIterator.php on line 62
PHP Deprecated:  Return type of Symfony\Component\Finder\Iterator\ExcludeDirectoryFilterIterator::accept() should either be compatible with FilterIterator::accept(): bool, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in phar:///Users/alex/.composer/vendor/spinupwp/spinupwp-cli/builds/spinupwp/vendor/symfony/finder/Iterator/ExcludeDirectoryFilterIterator.php on line 59

Deprecated: Return type of Symfony\Component\Finder\Iterator\ExcludeDirectoryFilterIterator::accept() should either be compatible with FilterIterator::accept(): bool, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in phar:///Users/alex/.composer/vendor/spinupwp/spinupwp-cli/builds/spinupwp/vendor/symfony/finder/Iterator/ExcludeDirectoryFilterIterator.php on line 59
PHP Deprecated:  Return type of Symfony\Component\Finder\Iterator\ExcludeDirectoryFilterIterator::hasChildren() should either be compatible with RecursiveIterator::hasChildren(): bool, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in phar:///Users/alex/.composer/vendor/spinupwp/spinupwp-cli/builds/spinupwp/vendor/symfony/finder/Iterator/ExcludeDirectoryFilterIterator.php on line 79

Deprecated: Return type of Symfony\Component\Finder\Iterator\ExcludeDirectoryFilterIterator::hasChildren() should either be compatible with RecursiveIterator::hasChildren(): bool, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in phar:///Users/alex/.composer/vendor/spinupwp/spinupwp-cli/builds/spinupwp/vendor/symfony/finder/Iterator/ExcludeDirectoryFilterIterator.php on line 79
PHP Deprecated:  Return type of Symfony\Component\Finder\Iterator\ExcludeDirectoryFilterIterator::getChildren() should either be compatible with RecursiveIterator::getChildren(): ?RecursiveIterator, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in phar:///Users/alex/.composer/vendor/spinupwp/spinupwp-cli/builds/spinupwp/vendor/symfony/finder/Iterator/ExcludeDirectoryFilterIterator.php on line 88

Deprecated: Return type of Symfony\Component\Finder\Iterator\ExcludeDirectoryFilterIterator::getChildren() should either be compatible with RecursiveIterator::getChildren(): ?RecursiveIterator, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in phar:///Users/alex/.composer/vendor/spinupwp/spinupwp-cli/builds/spinupwp/vendor/symfony/finder/Iterator/ExcludeDirectoryFilterIterator.php on line 88
PHP Deprecated:  Return type of Symfony\Component\Finder\Iterator\FilenameFilterIterator::accept() should either be compatible with FilterIterator::accept(): bool, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in phar:///Users/alex/.composer/vendor/spinupwp/spinupwp-cli/builds/spinupwp/vendor/symfony/finder/Iterator/FilenameFilterIterator.php on line 31

Deprecated: Return type of Symfony\Component\Finder\Iterator\FilenameFilterIterator::accept() should either be compatible with FilterIterator::accept(): bool, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in phar:///Users/alex/.composer/vendor/spinupwp/spinupwp-cli/builds/spinupwp/vendor/symfony/finder/Iterator/FilenameFilterIterator.php on line 31
PHP Deprecated:  Return type of Symfony\Component\Finder\Iterator\PathFilterIterator::accept() should either be compatible with FilterIterator::accept(): bool, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in phar:///Users/alex/.composer/vendor/spinupwp/spinupwp-cli/builds/spinupwp/vendor/symfony/finder/Iterator/PathFilterIterator.php on line 30

Deprecated: Return type of Symfony\Component\Finder\Iterator\PathFilterIterator::accept() should either be compatible with FilterIterator::accept(): bool, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in phar:///Users/alex/.composer/vendor/spinupwp/spinupwp-cli/builds/spinupwp/vendor/symfony/finder/Iterator/PathFilterIterator.php on line 30

In HandleExceptions.php line 97:
                                                          
  Call to undefined method Psr\Log\NullLogger::channel()
```

## Desired Outcome
Ideally the CLI would work.

## Steps to Reproduce

1. Install CLI
2. Run CLI
3. Notice the errors

## Related

- I have PHP and Composer installed via Homebrew on my M1 Mac running macOS 13.2 and always running the latest version of each, currently:

```
PHP 8.2.1 (cli) (built: Jan 12 2023 03:48:24) (NTS)
Copyright (c) The PHP Group
Zend Engine v4.2.1, Copyright (c) Zend Technologies
    with Xdebug v3.2.0, Copyright (c) 2002-2022, by Derick Rethans
    with Zend OPcache v8.2.1, Copyright (c), by Zend Technologies

Composer version 2.5.2 2023-02-04 14:33:22
```

## Acceptance Criteria

- The CLI runs as intended on any environment with any recent versions of PHP and Composer

**Comment by daankortenbach** [comment]
Same issue. Mac M1 macOS 13.2.1
```
In HandleExceptions.php line 97:
                                                          
  Call to undefined method Psr\Log\NullLogger::channel()  
                                                          

➜  ~ php -v
PHP 8.2.4 (cli) (built: Mar 16 2023 16:13:06) (NTS)
Copyright (c) The PHP Group
Zend Engine v4.2.4, Copyright (c) Zend Technologies
    with Zend OPcache v8.2.4, Copyright (c), by Zend Technologies
➜  ~ composer -v
   ______
  / ____/___  ____ ___  ____  ____  ________  _____
 / /   / __ \/ __ `__ \/ __ \/ __ \/ ___/ _ \/ ___/
/ /___/ /_/ / / / / / / /_/ / /_/ (__  )  __/ /
\____/\____/_/ /_/ /_/ .___/\____/____/\___/_/
                    /_/
Composer version 2.5.5 2023-03-21 11:50:05
```

"Fixed" by downgrading to php8.0 for my command line.

**Comment by georgefehr** [comment]
The latest version (v1.1.0) released today should support PHP 8.1 and PHP 8.2.
